libraries/AP_HAL_Linux/Perf.cpp

```diff
@@ -186,4 +195,322 @@ void Util::perf_count(perf_counter_t perf)
     perf_counter->count++;
 }
 
-#endif
+#ifdef HAVE_SYSTEMD
+
+#define DBUS_NAMESPACE "org.dronecode.Ardupilot"
+#define DBUS_PATH "/org/dronecode/ardupilot"
+
+PerfManager* PerfManager::_instance = nullptr;
+
+extern const sd_bus_vtable _perf_vtable[];
+
+extern "C" int _dbus_list_perf_callback(sd_bus_message *m, void *data, sd_bus_error *ret_error)
+{
+    PerfManager *perf_manager = (PerfManager *)data;
+    return perf_manager->dbus_list_perf_callback(m, data, ret_error);
+}
+
+extern "C" int _dbus_get_perf_callback(sd_bus_message *m, void *data, sd_bus_error *ret_error)
+{
+    PerfManager *perf_manager = (PerfManager *)data;
+    return perf_manager->dbus_get_perf_callback(m, data, ret_error);
+}
+
+static void* _dbus_thread_callback(void *data)
+{
+    PerfManager *perf_manager = (PerfManager *)data;
+    perf_manager->dbus_process_callback();
+    return nullptr;
+}
+
+PerfManager::PerfManager()
+{
+    int r = sd_bus_open_system(&_bus);
+    if (r < 0) {
+        AP_HAL::panic("PerfManager: Unable to open system bus.");
+    }
+
+    r = sd_bus_request_name(_bus, DBUS_NAMESPACE, 0);
+    if (r < 0) {
+        sd_bus_unref(_bus);
+        AP_HAL::panic("PerfManager: Unable to request bus name: "
+                      DBUS_NAMESPACE ", probably you need add the "
+                      "configuration file to /etc/dbus-1/system.d");
+    }
+
+    r = sd_bus_add_object_vtable(_bus, &_slot,
+                                 DBUS_PATH,
+                                 DBUS_NAMESPACE ".Perf",
+                                 _perf_vtable,
+                                 this);
+    if (r < 0) {
+        sd_bus_unref(_bus);
+        AP_HAL::panic("PerfManager: Unable to register Perf interface.");
+    }
+
+    r = pthread_create(&_thread, nullptr, _dbus_thread_callback, this);
```

Comment:
humn... if you do this, the thread will have the same (rt) prio as the main thread. I.e. it can starve the other threads.  You can check the thread priorities with:

`ps -eL -o tid,comm,class,rtprio $(pidof arducopter-quad)`, assuming you are running arducopter-quad

Please extend the Thread interface to allow setting the prio for